packages/relay/src/lib/eth.ts

```diff
@@ -2208,9 +2210,16 @@ export class EthImpl implements Eth {
 
   private async getAccountNonceForHistoricBlock(
     address: string,
-    blockNum: number,
+    blockNumOrHash: number | string,
     requestIdPrefix?: string,
   ): Promise<string> {
+    const getBlock = await this.mirrorNodeClient.getBlock(blockNumOrHash);
```

Comment:
Q: do we need to get the block each time?
Please make sure in the flow we're not adding a new mirror node call each time that's no always needed